src/libguac/socket.c

```diff
@@ -286,56 +308,70 @@ ssize_t __guac_socket_write_base64_triplet(guac_socket* socket,
         output[3] = '=';
     }
 
-    /* At this point, 4 base64 bytes have been written */
-    if (guac_socket_write(socket, output, 4))
-        return -1;
-
-    /* If no second byte was provided, only one byte was written */
-    if (b < 0)
-        return 1;
-
-    /* If no third byte was provided, only two bytes were written */
-    if (c < 0)
-        return 2;
+    return 0;
+}
 
-    /* Otherwise, three bytes were written */
-    return 3;
+ssize_t guac_socket_flush_base64(guac_socket* socket) {
+    const unsigned char* src = socket->__ready_buf;
 
-}
+    int encodedCount = 0;
+    int remaining = socket->__ready;
 
-ssize_t __guac_socket_write_base64_byte(guac_socket* socket, int buf) {
+    // Encode bytes in groups of three
+    while(remaining > 2) {
+        __guac_socket_encode_base64(src[0], src[1], src[2], socket->__encoded_buf + encodedCount);
 
-    int* __ready_buf = socket->__ready_buf;
+        remaining -= 3;
+        src += 3;
+        encodedCount += 4;
+    }
 
-    int retval;
+    // Take care of partial remnants
+    if(remaining == 2) {
+        __guac_socket_encode_base64(src[0], src[1], -1, socket->__encoded_buf + encodedCount);
+        encodedCount += 4;
+    } else if(remaining == 1) {
```

Comment:
1. Space between `else if` and `(` please (see surrounding code).
2. Please do not cuddle the `else if` (see https://guacamole.apache.org/guac-style/#braces)